### ISSUE TITLE
Load error when trying to run `bundle`

### ISSUE THREAD

**Issue by PwnKitteh** [jekyll/jekyll#6996]
- [x] I am on (or have tested on) ***Debian/Ubuntu*** GNU/Linux
- [x] I was trying to install.

## My Reproduction Steps

I installed jekyll this morning, following the steps for installation on ubuntu linux. When I came back to it today, I got the following error message whever I run `bundle`: 

```
/home/jacob/gems/bin/bundle:23:in `load': cannot load such file -- /usr/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/exe/bundle (LoadError)
        from /home/jacob/gems/bin/bundle:23:in `<main>'
```

I've been installing and reinstalling, reading countless stack overflow and github issue threads, and none of the solutions work for me. I'm a complete ruby noob, so I'm not quite sure what logs you want. `bundler` works, and it says I'm up to date. Trying to run `jekyll` returns 

```
/usr/local/lib/site_ruby/2.3.0/bundler/runtime.rb:313:in `check_for_activated_spec!': You have already activated i18n 1.0.1, but your Gemfile requires i18n 0.9.5. Prepending `bundle exec` to your command may solve this. (Gem::LoadError)
        from /usr/local/lib/site_ruby/2.3.0/bundler/runtime.rb:31:in `block in setup'
        from /usr/local/lib/site_ruby/2.3.0/bundler/runtime.rb:26:in `map'
        from /usr/local/lib/site_ruby/2.3.0/bundler/runtime.rb:26:in `setup'
        from /usr/local/lib/site_ruby/2.3.0/bundler.rb:107:in `setup'
        from /home/jacob/gems/gems/jekyll-3.8.1/lib/jekyll/plugin_manager.rb:50:in `require_from_bundler'
        from /home/jacob/gems/gems/jekyll-3.8.1/exe/jekyll:11:in `<top (required)>'
        from /home/jacob/gems/bin/jekyll:22:in `load'
        from /home/jacob/gems/bin/jekyll:22:in `<main>'
```

My gems are 

```
Gems included by the bundle:
  * activesupport (5.2.0)
  * addressable (2.5.2)
  * bundler (1.16.0)
  * colorator (1.1.0)
  * concurrent-ruby (1.0.5)
  * em-websocket (0.5.1)
  * eventmachine (1.2.6)
  * faraday (0.15.0)
  * ffi (1.9.23)
  * forwardable-extended (2.6.0)
  * gemoji (3.0.0)
  * html-pipeline (2.8.0)
  * http_parser.rb (0.6.0)
  * i18n (0.9.5)
  * jekyll (3.8.1)
  * jekyll-data (1.0.0)
  * jekyll-feed (0.9.3)
  * jekyll-gist (1.5.0)
  * jekyll-paginate (1.1.0)
  * jekyll-sass-converter (1.5.2)
  * jekyll-sitemap (1.2.0)
  * jekyll-watch (2.0.0)
  * jemoji (0.10.0)
  * kramdown (1.16.2)
  * liquid (4.0.0)
  * listen (3.1.5)
  * mercenary (0.3.6)
  * mini_portile2 (2.3.0)
  * minimal-mistakes-jekyll (4.11.2)
  * minitest (5.11.3)
  * multipart-post (2.0.0)
  * nokogiri (1.8.2)
  * octokit (4.8.0)
  * pathutil (0.16.1)
  * public_suffix (3.0.2)
  * rake (10.5.0)
  * rb-fsevent (0.10.3)
  * rb-inotify (0.9.10)
  * rouge (3.1.1)
  * ruby_dep (1.5.0)
  * safe_yaml (1.0.4)
  * sass (3.5.6)
  * sass-listen (4.0.0)
  * sawyer (0.8.1)
  * thread_safe (0.3.6)
  * tzinfo (1.2.5)
```

## The Output I Wanted

jekyll to work after installation

This worked out of the box the last time I installed jekyll a couple of months ago, but I had to reinstall on my machine because I managed to mess up python and now it's not working. Either I'm a complete idiot (which is probable) or something changed in the last 3 months which might be causing this. 
I appreciate any and all help. I know this is more of a ruby than a jekyll issue, so I'm sorry for using you for ruby support. I'm stuck, and since the only reason I'm using ruby is for jekyll, I figured I'd turn here. 
Thanks.

**Comment by ashmaroli** [comment]
Have you tried `.. Prepending `bundle exec` to your command..`?

**Comment by ashmaroli** [comment]
Have you tried `.. Prepending 'bundle exec' to your command..`?

**Comment by uberfuzzy** [comment]
tldr; type `bundle exec jekyll`


`bundler` installs the gems listed in your Gemfile to create a little "pocket dimension" of only the gems from the Gemfile, that are separate from anything else on your system. To get jekyll to use those gems that it installed, it also needs to run inside that "pocket", so you must tell bunder to run it for you, using `bundler exec` just like the error message says.

It is failing because you are running jekyll on your system (that doesnt have the correct gems installed at the system level), and not inside the bundler bubble.

It worked "out of the box" last time because you likely just did `gem install jekyll`. 

You are mixing paradigms. You can run jekyll at the system level, or with bundler and a Gemfile. Just dont mix them. Hint: using bundler is strongly advised, as it can manage dependencies and relationships and updates far easier.

**Comment by PwnKitteh** [comment]
I'm sorry for re-opening this, if I wasn't clear enough, but the `bundle` command is broken on my system. Running `bundle exec jekyll` returns `/home/jacob/gems/bin/bundle:23:in `load': cannot load such file -- /usr/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/exe/bundle (LoadError)
        from /home/jacob/gems/bin/bundle:23:in `<main>'`

Thanks for the reply though; I appreciate it

**Comment by DirtyF** [comment]
Have you read https://jekyllrb.com/tutorials/using-jekyll-with-bundler ?

**Comment by ashmaroli** [comment]
Apart from following the above-mentioned documentation, you can simply try "resetting" Bundler.

The "bundle" currently has Bundler-1.16.0 but the error message seems to be from Bundler-1.16.1

What you can do is try again after deleting your `Gemfile.lock`. This should force Bundler to re-resolve dependencies.

Running `bundle update` will alternatively do the same thing but will also install the highest possible versions of gems that are compatible with the other gems listed in your `Gemfile`.

If the same error repeats, simply *reinstall Bundler*

```
gem uninstall bundler
```
```
gem install bundler
```
